docs/reference-docs/auth-backend-and-frontend.md

```diff
@@ -0,0 +1,228 @@
+    PROCESS_RETRY = '/orchestrator/processes/retry/', // called when determining if the button to trigger a process retry should be shown
+    PROCESS_USER_INPUT = '/orchestrator/processes/user-input/', // called when determining if th
+    SUBSCRIPTION_CREATE = '/orchestrator/processes/create/process/menu', // called when determining if create if actions that trigger a create workflow should be displayed
+    SUBSCRIPTION_MODIFY = '/orchestrator/subscriptions/modify/', // called when determining if create if actions that trigger a modify workflow should be displayed
+    SUBSCRIPTION_TERMINATE = '/orchestrator/subscriptions/terminate/', // called when determining if create if actions that trigger a terminate workflow should be displayed
+    SUBSCRIPTION_VALIDATE = '/orchestrator/subscriptions/validate/', // called when determining if create if actions that trigger a validate task should be displayed
+    TASKS_CREATE = '/orchestrator/processes/create/task', // called when determining if create if actions that trigger a task should be displayed
+    TASKS_RETRY_ALL = '/orchestrator/processes/all-tasks/retry', // called when determining if create if actions that trigger retry all tasks task should be displayed
+    SETTINGS_FLUSH_CACHE = '/orchestrator/settings/flush-cache', // called when determining if a button to flush cache should be displayed
+    SET_IN_SYNC = '/orchestrator/subscriptions/set-in-sync', // called when determining if a button to set a subscription in sync should be displayed
+}
+```
+
+Note: Components that are hidden for unauthorized users are still part of the frontend application, authorization just makes sure
+unauthorized users are not presented with actions they are not allowed to take. The calls these actions
+make can still be made through curl calls for example. Additional authorization needs to be implemented on these calls on the backend.
+
+### Backend
+
+**ENV variables**
+These variables need to be set for authentication to work on the backend
+
+```
+...
+# OIDC settings
+OAUTH2_ACTIVE: bool = True
+OAUTH2_AUTHORIZATION_ACTIVE: bool = True
+OAUTH2_RESOURCE_SERVER_ID: str = ""
+OAUTH2_RESOURCE_SERVER_SECRET: str = ""
+OAUTH2_TOKEN_URL: str = ""
+OIDC_BASE_URL: str = ""
+OIDC_CONF_URL: str = ""
+
+# OPtional OPA settings
+OPA_URL: str = ""
+```
+
+With the variables provided requests to endpoints will return 403 errorcodes for users that are not logged in and 401 error codes for users that are not authorized for a call or part of a call
+
+#### Customization
+
+When initiating the `OrchestratorCore` class it's [`auth_manager`][6] property is set to `AuthManager`. AuthManager is provided by [oauth2_lib][7].
+
+There are 3 methods from `AuthManager` that are used for authentication and authorization. `AuthManager` provides defaults and methods to override
+the set custom fuctions to override the default implementations. The default implementations are:
+
```

Comment:
I had to implement one function to get authentication working.

I can share it, if you want; it handles the token decode/check with a "almost standard" keycloak.